### PR TITLE
Clang-format all files (.cpp, .h, .cu)

### PR DESCRIPTION
**Pull request by manuelF**
Just like in #109, run the tool with the changes since https://github.com/MALBECC/lio/pull/160

```
cd g2g
find . | grep -E "\.(h|cpp|cu)$" | xargs clang-format-10 -i -style=file
```
This PR might bring a lot of conflicts to others, but the solution is to run the command above *before you merge*. This should cause your diffs to be zero.

Cuda changes are now correctly formatted (after clang added support https://reviews.llvm.org/D42589) but there are a *lot*. They are all auto-generated but if people reformat-before-merge before there should be no problem.

After fixing the formatting, `cpp-lint` shows the following issues:
https://github.com/manuelF/lio/runs/4218078570?check_suite_focus=true

Likely not all of them are worth fixing, so the tool can be configured to ignore certain kind of issues, like copyright. 
(see https://android.googlesource.com/platform/art/+/master/CPPLINT.cfg for example)

I will configure it independently in a separate PR.

### CODE REVIEW

**Comment by fedepedron** [comment]
Hmm this one might need more consensus. Opinions?
@nanolebrero @gonzalodm @ramirezfranciscof @charlyqchm

**Comment by ramirezfranciscof** [comment]
What is this? Did I step on a time machine?

My opinion is: lets refactor the codebase into python while the argentinians are sleeping.

(On a more serious note, I'm a little rusty on this so a couple of questions: I only see changes on the sourcefiles, so the linter was already configured in the `.git` files? I don't remember anything triggering on commits. Also, is there an automatic formatter for these files?)

**Comment by manuelF** [comment]
Hi, all of this is automatically generated via the compiler (clang), that enables a standard formatting in the codebase. This was done in https://github.com/MALBECC/lio/pull/160 but it didn't include cuda files then, as the formatter did not support it correctly. Now it does.

The automatic formatter-checker will now be enabled in order then to not have any other regressions: https://github.com/MALBECC/lio/pull/364 (only when trying to merge onto develop and master, as to allow people to freely experiment and not getting any complaints)

Basically this just standarizes the entire codebase (C++/Cuda for now), old and new, into a single unified style.

**Comment by manuelF** [comment]
Checked that after this PR all the code is clang-format compliant. New or modified C++ and CUDA code will be checked against style to prevent regressions (only on `master` and `develop` branches).

**Comment by manuelF** [comment]
Abandoning this PR as this will probably never get merged.
